### DISCUSSION TITLE
How to set single ended inequality constraint ?

### DISCUSSION THREAD

**Discussion by equation33** [osqp/osqp#661] [Embedded Optimization]
Hi Expert.

May I ask how to set the single ended inequality constraint? 

In the reference, the inequality constraint can be set as l<=A*x <= u.

When my constraint is only A*x <= u, how to set that ? 

I tried to set l = -1000 as the minus infinity value. It works well, but I want to know if there is any better setting method? 

Thanks very much.

**Comment by imciner2** [comment]
You can set single ended constraints by making the entry in $l$ or $u$  be `-OSQP_INFTY` or `OSQP_INFTY`, respectively. (that is actually just a really big number as well).

**Reply by equation33** [reply]
Thanks very much.  OSQP_INFTY is defined as 1e30. Got it.

**Reply by imciner2** [reply]
It is either 1e30 or 1e17, depending on whether the library is using double or float for its floating point type, so using the constant is a bit more portable across library configs.
